# Graded assignment 1

Mark Klik & Misja Mikkers


# Packages

We need the package `tidyverse`

In [7]:
library(tidyverse) 

# Assignment 1

First, we generate some random data:

In [8]:
df1 <- as.data.frame(matrix(runif(20*2), ncol=20) )
head(df1)

V1        V2        V3         V4        V5        V6        V7        
1 0.4839952 0.3758082 0.02284212 0.7229194 0.8155349 0.7706172 0.08837577
2 0.2900821 0.8590679 0.08815183 0.5356759 0.6312861 0.9520740 0.94567104
  V8        V9          V10       V11       V12       V13       V14      
1 0.3135331 0.935104482 0.2213972 0.1252009 0.8067264 0.2678427 0.4356793
2 0.8978328 0.008245861 0.7508925 0.1486317 0.5135251 0.1358131 0.1172746
  V15       V16       V17       V18       V19       V20      
1 0.4705297 0.5475941 0.2579173 0.3293451 0.9423829 0.4388025
2 0.7180712 0.1613425 0.7318326 0.2202487 0.2496792 0.3159079

_df1_ is a wide dataframe. The goal of the assignment is to change _df1_ into a long dataframe _df2_. And show the first rows of the dataframe using `head(df2)`

The dataframe _df2_ should look like this:

| Key | Value      |
|-----|------------|
| V1  | 0.09413864 |
| V1  | 0.69782526 |
| V2  | 0.23460454 |
| V2  | 0.51520514 |
| V3  | 0.63181844 |
| V3  | 0.22807211 |

In [9]:

df2 <- gather (df1)
head(df2)


key value     
1 V1  0.48399517
2 V1  0.29008206
3 V2  0.37580820
4 V2  0.85906792
5 V3  0.02284212
6 V3  0.08815183

## Assignment 2

We will generate some data:

In [10]:
df3 <- as.data.frame(read.table(header=TRUE, text='
      id        V_1       V_2                
          1  42.467573  -87.810047     
          2  42.049148  -88.273029
          3  39.110539  -90.324080
                       '))

head(df3)

id V_1      V_2      
1 1  42.46757 -87.81005
2 2  42.04915 -88.27303
3 3  39.11054 -90.32408

In [11]:
df4 <- as.data.frame(read.table(header=TRUE, text='
     V_3            V_4         id                 
      one         1h            1
      two         1h            1
      three       8h            2
      four        1h            4
'))

head(df4)

V_3   V_4 id
1 one   1h  1 
2 two   1h  1 
3 three 8h  2 
4 four  1h  4


You will have to merge these files, returning all rows and all columns from both _df3_ and _df5_. Where there are not matching values, return _NA_ for the missings. 

This should be the end result:

| id | V_1      | V_2       | V_3   | V_4 |
|----|----------|-----------|-------|-----|
| 1  | 42.46757 | -87.81005 | one   | 1h  |
| 1  | 42.46757 | -87.81005 | two   | 1h  |
| 2  | 42.04915 | -88.27303 | three | 8h  |
| 3  | 39.11054 | -90.32408 | NA    | NA  |
| 4  | NA       | NA        | four  | 1h  |



In [12]:
df5 <- full_join(df3,df4,by="id")
df5

id V_1      V_2       V_3   V_4
1 1  42.46757 -87.81005 one   1h 
2 1  42.46757 -87.81005 two   1h 
3 2  42.04915 -88.27303 three 8h 
4 3  39.11054 -90.32408 NA    NA 
5 4        NA        NA four  1h

End of notebook